Raise a warning when applying a pulse gradient transform to a QNode directly (#4241)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Improvements 🛠</h3>

> * The pulse differentiation methods, `pulse_generator` and `stoch_pulse_grad` now raise a warning when they
>   are applied to a `QNode` directly, which is only supported for scalar pulse parameters. For non-scalar
>   pulse parameters, use differentiation via a JAX entry point (`jax.grad`, `jax.jacobian`, ...) instead
>   [(4241)](https://github.com/PennyLaneAI/pennylane/pull/4241)

Reviewer: ```suggestion
* The pulse differentiation methods, `pulse_generator` and `stoch_pulse_grad` now raise an error when they
  are applied to a `QNode` directly. Instead, use differentiation via a JAX entry point (`jax.grad`, `jax.jacobian`, ...).
  [(4241)](https://github.com/PennyLaneAI/pennylane/pull/4241)
```
